src/Condition.js

```diff
@@ -0,0 +1,233 @@
+ * governing permissions and limitations under the License.
+ */
+
+const url = require('url');
+
+// To avoid forward referencing the transformer function
+let transform;
+
+const Affix = {
+  PREFIX: 1,
+  INFIX: 2,
+};
+
+/**
+ * Boolean conditions
+ */
+const booleanConditionMap = {
+  or: {
+    name: 'or',
+    type: Affix.INFIX,
+    op: ' || ',
+  },
+  and: {
+    name: 'and',
+    type: Affix.INFIX,
+    op: ' && ',
+  },
+  not: {
+    name: 'not',
+    type: Affix.PREFIX,
+    op: ' !',
+  },
+};
+
+class BooleanCondition {
+  constructor(condition, children) {
+    this._condition = condition;
+
+    switch (condition.type) {
+      case Affix.PREFIX:
+        this._children = transform(children);
+        break;
+      default:
+        this._children = children.map(child => transform(child));
+        break;
+    }
+  }
+
+  toVCL() {
+    let clause;
+
+    switch (this._condition.type) {
+      case Affix.PREFIX:
+        clause = this._children.toVCL();
+        return `${this._condition.op}(${clause})`;
+      default:
+        clause = this._children.map(child => child.toVCL()).join(this._condition.op);
+        return `(${clause})`;
+    }
+  }
+
+  evaluate(req) {
+    switch (this._condition.name) {
+      case 'and':
+        return this._children.reduce((result, child) => result && child.evaluate(req), true);
+      case 'or':
+        return this._children.reduce((result, child) => result || child.evaluate(req), false);
+      default:
+        return !this._children.evaluate(req);
+    }
+  }
+}
+
+/**
+ * PropertyCondition
+ */
+class PropertyCondition {
+  constructor(name, type, op, value, vcl, express) {
+    this._name = name;
+    this._type = type;
+    this._op = op;
+    this._value = value;
+    this._vcl = vcl;
+    this._express = express;
+  }
+
+  toVCL() {
+    const quote = this._type === 'string' ? '"' : '';
+    let value = this._value;
+    let op = this._op;
+
+    if (quote === '"' && !op) {
+      // substring-start
+      value = `^${value}`;
+      op = '~';
+    }
+    if (!op || op === '=') {
+      // operand defaults to equal
+      op = '==';
+    }
+    const vcl = typeof this._vcl === 'function' ? this._vcl(this) : this._vcl;
+    return `${vcl} ${op} ${quote}${value}${quote}`;
+  }
+
+  evaluate(req) {
+    const actual = this._express(req, this);
+    const value = this._value;
+
+    if (!actual) {
+      return false;
+    }
+
+    switch (this._op) {
+      case '=':
+        return actual === value;
+      case '~':
+        return !!actual.match(value);
+      case '<':
+        return actual < value;
+      case '>':
+        return actual > value;
+      default:
+        return actual.startsWith(value);
+    }
+  }
+
+  get name() {
+    return this._name;
+  }
+
+  get type() {
+    return this._type;
+  }
+}
+
+/**
+ * Known properties
+ */
+const propertyConditionMap = {
+  url: {
+    vcl: 'req.url',
+    express: req => req.path,
```

Comment:
this is not correct. the `url` property should be `${scheme}://${req.headers.host}${req.fullUrl}`; similar in VCL.
the problem is probably the `scheme`...don't know if this is available in the request or in fastly.

Reply:
Right, I'm actually fixing it now... 😄

Reply:
It's called `req.protocol` in Fastly